### DISCUSSION TITLE
What's the correct way to style the LineNumberMargin?

### DISCUSSION THREAD

**Discussion by Max-H-0** [AvaloniaUI/AvaloniaEdit#395] [Q&A] [answered]
I'd like to get this:
![image](https://github.com/AvaloniaUI/AvaloniaEdit/assets/81490420/a405f94c-96bc-4a23-9d58-bd2712f8f223)
To look more like this:
![image](https://github.com/AvaloniaUI/AvaloniaEdit/assets/81490420/7a336a08-1759-4b19-a848-ffa410a588a1)

I tried it like this
```xaml
    <Style Selector="ae|TextEditor">
        <Style Selector="^ > ItemsControl">
            <Setter Property="Width" Value="80"/>
            
            <Style Selector="^ > aee|LineNumberMargin">
                <Setter Property="Width" Value="30"/>
            </Style>
            
            <Style Selector="^ >Line">
                <Setter Property="Margin" Value="25,0,2,0"/>
            </Style>
        </Style>
    </Style>
```
(also with /template/) 
but couldn't get anything to change

Is this even the correct way to achieve what I'm going for?
In case it is, what are the correct selectors?

**Comment by NickRimmer** [comment, marked as answer]
Dirty solution to override vertical line, but this is a hack and can stop working with future Editor updates

```c#
if (_textEditor.TextArea.LeftMargins.Count >= 2 && _textEditor.TextArea.LeftMargins[1] is Line)
{
    var line = new Line
    {
        StartPoint = new Point(0, 0),
        EndPoint = new Point(0, 1),
        StrokeDashArray = new AvaloniaList<double> { 0, 2 },
        Stretch = Stretch.Fill,
        StrokeThickness = 1,
        StrokeLineCap = PenLineCap.Round,
        Margin = new Thickness(10, 0, 20, 0),
        Tag = Tag
    };
    
    _textEditor.TextArea.LeftMargins.RemoveAt(1);
    _textEditor.TextArea.LeftMargins.Insert(1, line);
    
    var lineNumbersForeground = new Binding("LineNumbersForeground") { Source = _textEditor };
    line.Bind(Shape.StrokeProperty, lineNumbersForeground);
}
```

Tested with Demo project

![image](https://github.com/AvaloniaUI/AvaloniaEdit/assets/1754265/f1e474b9-bdb9-4a12-bc86-d5bfd181413c)

**Reply by NickRimmer** [reply]
I added PR to set margins easier (;
#402

**Reply by Max-H-0** [reply]
Thank you!

**Comment by nullptr-w8** [comment]
<Style Selector="AvaloniaEdit|TextEditor Panel">
		<Setter Property="Margin" Value="20,0,0,0"/>
	</Style>

	<Style Selector="AvaloniaEdit|TextEditor StackPanel">
		<Setter Property="Spacing" Value="5"/>
	</Style>
